testdata/p4_16_samples_outputs/forloop2.p4-stderr

```diff
@@ -0,0 +1,2 @@
+[--Wwarn=uninitialized_use] warning: retval may be uninitialized
```

Comment:
The warning is misleading. First of all, there is already a value returned. Second, there is no `retval` in the source code. `retval` is just internal symbol name inserted during lowering and therefore should not be exposed.

Reply:
Yes, I'm not sure where the warning is coming from -- I'll look into it.

Reply:
I think these are connected (the previous `hasReturned` thing). Downstream we do not have these, however, we are having some decent amount of fixes (and we do not rely on `hasReturn` in general). Though, I have not yet found what was the fix and why we have not upstreamed it...

Reply:
The warning comes from a combination of the code produced by RemoveReturns and the flow analysis code -- the latter does not do value correlation, so it finds an impossible path through the code where it thinks the `retval` introduced by RemoveReturns might be uninitialized.  Improving flow analysis to also do value analysis (maybe just for `bool` vars) would be major work.

We could have RemoveReturns always initialize `retval`.  Or we could have it mark retval with an annotation that would supress the warning.

Reply:
This problem doesn't have anything specific to do with loops -- it can happen for any function that contains multiple `return` statements that gets inlined.